feat: add icons and previews for docs external and glossary links

saimonkat opened the pull request:
This PR adds icons and previews for docs external and glossary links

## External links
- arrow icon added for ALL external links in docs content
- preview is customizable for every link and can be implemented with 
  `<LinkPreview href="" title="" preview="">text</LinkPreview>`

![image](https://github.com/user-attachments/assets/c2565107-42fc-470f-8c59-51fc13fdbc20)

### Steps to test
- check if all external links have arrow icon
- check if `<LinkPreview>` component works well
- check dark and light themes

## Glossary links
- book icon added for ALL glossary links in docs content
- preview is automatically generated from the glossary section and truncated to **3 lines**

![image](https://github.com/user-attachments/assets/e0235704-7424-4792-8fea-1c37bd7e4181)

### Steps to test
- check if all glossary links have book icon
- test removing markdown formatting from preview (links, bold and italic text styles, lists, code blocks, inline code, tables e.t.c)
- check if preview text  truncated to **3 lines** and ends with `...` for longer texts
- check dark and light themes

### TO-DO
- [x] remove test `<LinkPreview>` and `glossary` links from docs before release

[Preview](https://neon-next-git-docs-external-links-neondatabase.vercel.app/docs/introduction)

danieltprice commented on content/docs/introduction.md:
```diff
@@ -3,7 +3,7 @@ title: Neon documentation
 ---
 
-Neon is a serverless Postgres platform designed to help you build reliable and scalable applications faster. We separate compute and storage to offer modern developer features such as autoscaling, branching, point-in-time restore, and more. Get started today with our [generous free plan](https://console.neon.tech).
+Neon is a serverless Postgres platform designed to help you build reliable and scalable applications faster. We separate compute and storage to offer modern developer features such as autoscaling, branching, point-in-time restore, and more. Get started today with our <LinkPreview href="https://console.neon.tech" title="PostgreSQL" preview="PostgreSQL is an open source object-relational database s ...
```
```suggestion
Neon is a serverless Postgres platform designed to help you build reliable and scalable applications faster. We separate compute and storage to offer modern developer features such as autoscaling, branching, point-in-time restore, and more. Get started today with our [generous free plan](https://console.neon.tech).
```

saimonkat replied:
Hi @danieltprice thank you your review! How do you suggest to customize previews titles and texts for external links? 

We discussed this task with Neon team and suggested 2 options: custom component `<LinkPreview>` with props for titles and text for preview, or a separate json file to keep all the data there. So the team chose the first option and we went along with it. 

Do you think a custom json file would be better?

saimonkat replied:
@danieltprice may I push our updates for Glossary links right in this PR as well?

danieltprice reviewed (approved):
This looks great. Thank you!
I've changed the link on the main page. We'll revisit to start adding tooltips.

saimonkat commented:
> I've adjusted changed the link on the main page. We'll revisit to start adding tooltips.

Oh I see got it! Right, the custom link was just for the review and presentation, I'll revert it before the release for sure!

saimonkat commented:
Hi @danieltprice we've added Glossary links previews to this PR

We've added support for cleaning markdown formatting from preview, which may include links, bold and italic text styles, lists, code blocks, inline code, tables e.t.c. Please approve if it's correct

![image](https://github.com/user-attachments/assets/ff7b2775-3570-4c25-a275-a35dec76bb6a)

![image](https://github.com/user-attachments/assets/9d056aed-9892-40f2-8833-3b9604a99d4b)

danieltprice commented:
@saimonkat Thank you! The Markdown formatting looks good, as far as I can tell. I clicked through several topics, and I did not see any issues.

Regarding the 64-character limit for the tooltip, it looks like we're only using about half of the second line. Would it make sense to increase that character limit a bit to use the second line a little more fully? I chose 64 because it's what Vercel appeared to use as a limit in their tooltips.

saimonkat commented:
@danieltprice another quick question: do we need to handle this type of links in Glossary?

![image](https://github.com/user-attachments/assets/e9d60975-0749-47be-bf96-1186911f51e1)

I mean it looks like the typo and not even rendering
We already have the handler for the classic md links: `[root branch](#root-branch)`

danieltprice commented:
@saimonkat 
1. Three full lines would be great.
2. I would say ignore those types of links 

Thank you! Looking forward to merging this!
